ui/ui-frontend/projects/referential/src/app/audit/audit-create/audit-create.component.ts

```diff
@@ -100,24 +100,8 @@ export class AuditCreateComponent implements OnInit {
   }
 
   ngOnInit() {
-    this.externalParameterService.getUserExternalParameters().subscribe((parameters) => {
-      const accessConctractId: string = parameters.get(ExternalParameters.PARAM_ACCESS_CONTRACT);
-      if (accessConctractId && accessConctractId.length > 0) {
-        this.accessContractId = accessConctractId;
-        this.auditService.getAllAccessionRegister(this.accessContractId).subscribe((accessionRegisterSummaries) => {
-          this.accessionRegisterSummaries = accessionRegisterSummaries;
-        });
-      } else {
-        this.snackBar.open(
-          $localize`:access contrat not set message@@accessContratNotSetErrorMessage:Aucun contrat d'accès n'est associé à l'utiisateur`,
-          null,
-          {
-            panelClass: 'vitamui-snack-bar',
-            duration: 10000,
-          }
-        );
-      }
-    });
+    this.externalParameterService.getUserExternalParameters()
+      .subscribe((parameters) => this.setAccesContractIdAndAccessionRegisterSummaries(parameters));
```

Comment:
this.setAccesContractIdAndAccessionRegisterSummaries(parameters) contient un subscribe ce qui est anti pattern 
use les operateur de rxjs à la place 
https://www.thinktecture.com/en/angular/rxjs-antipattern-1-nested-subs/

Reply:
yes, je vais voir si je peux changer ca facilement